sys/walltime/walltime.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2025 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_walltime
+ * @{
+ *
+ * @file
+ * @brief       Wall-Clock time access
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+#include "auto_init.h"
+#include "auto_init_utils.h"
+#include "auto_init_priorities.h"
+
+#include "periph/rtc.h"
+#include "rtc_utils.h"
+#include "walltime.h"
+#include "ztimer.h"
+
+#if !defined(MODULE_PERIPH_RTC) && !defined(MODULE_RTT_RTC)
+#define ZTIMER_FALLBACK
+#endif
+
+static uint32_t _boottime;
+#ifdef BACKUP_RAM
+static uint32_t _boottime_bkup;
+#endif
+#ifdef ZTIMER_FALLBACK
+uint32_t ztimer_offset;
```

Comment:
even though i rarely run my devboards for 50 days or longer there should be a "mitigation for the 49day overflow"
eg a overflow counter

Reply:
ah here: `uint32_t now = ztimer_now(ZTIMER_MSEC);` you mean.